py3status/modules/usbguard.py

```diff
@@ -200,182 +82,94 @@ def post_config_hook(self):
+                composite = self.py3.safe_format(getattr(self, btn), device)
+                device[btn] = self.py3.composite_update(
+                    composite,
+                    {"index": "{}/{}".format(device["id"], btn.split("_")[-1])},
                 )
 
-            device_composite = self.py3.safe_format(self.format_device, device)
-            if index == self.device_index:
-                format_device_list = device_composite
-
-            new_device.append(device_composite)
+            device_info.append(self.py3.safe_format(self.format_device, device))
 
         format_device_separator = self.py3.safe_format(self.format_device_separator)
-        format_device = self.py3.composite_join(format_device_separator, new_device)
-
-        return format_device, format_device_list
+        format_device = self.py3.composite_join(format_device_separator, device_info)
 
-    def _notify_user(self, device):
-        format_notification_message = self.py3.safe_format(
-            self.format_notification_message, device
-        )
-        format_notification_title = self.py3.safe_format(
-            self.format_notification_title, device
-        )
-        self.py3.notify_user(
-            msg=format_notification_message,
-            title=format_notification_title,
-            icon="/usr/share/icons/hicolor/scalable/apps/usbguard-icon.svg",
-        )
+        return format_device
 
     def usbguard(self):
-        filter_name = self.filters[self.filter_index]
-        device_data = self._get_devices(filter_name)
-        format_device, format_device_list = self._manipulate_devices(
-            device_data, filter_name
-        )
-
-        if self.device_index > self.device_length:
-            self.device_index = self.device_length
-
-        format_button_device = self.py3.safe_format(
-            self.format_button_device,
-            {"device": self.device_index, "devices": self.device_length},
-        )
-        self.py3.composite_update(format_button_device, {"index": "button_device"})
-
-        format_button_filter = self.py3.safe_format(
-            self.format_button_filter, {"filter": filter_name}
-        )
-        self.py3.composite_update(format_button_filter, {"index": "button_filter"})
-        format_button_permanent = self.py3.safe_format(self.format_button_permanent)
-        self.py3.composite_update(
-            format_button_permanent, {"index": "button_permanent"}
-        )
+        self.devices = self._get_devices()
 
         usbguard_data = {
-            "device": len(device_data),
-            "format_button_filter": format_button_filter,
-            "format_button_permanent": format_button_permanent,
-            "format_device": format_device,
-            "format_button_device": format_button_device,
-            "format_device_list": format_device_list,
+            "device": len(self.devices),
+            "format_device": self._format_device(self.devices),
```

Comment:
Maybe you don't need `self.` part. Is contained here?

Reply:
You are right, updated 👍